cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/IOUContract.kt

```diff
@@ -30,24 +30,35 @@ class IOUContract : Contract {
      * considered valid.
      */
     override fun verify(tx: LedgerTransaction) {
-        val command = tx.commands.requireSingleCommand<Commands.Create>()
-        requireThat {
-            // Generic constraints around the IOU transaction.
-            "No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
-            "Only one output state should be created." using (tx.outputs.size == 1)
-            val out = tx.outputsOfType<IOUState>().single()
-            "The lender and the borrower cannot be the same entity." using (out.lender != out.borrower)
-            "All of the participants must be signers." using (command.signers.containsAll(out.participants.map { it.owningKey }))
-
-            // IOU-specific constraints.
-            "The IOU's value must be non-negative." using (out.value > 0)
+        val command = tx.commands.requireSingleCommand<Commands>()
+
+        when (command.value) {
+            is Commands.Create -> requireThat {
+                // Generic constraints around the IOU transaction.
+                "No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
```

Comment:
Still preventing money from being exchanged at Create time.

Reply:
I didn't touch `Create` at all. I left it as is, because base on task description from module3 I understand, that I need to
> make it possible to cancel an IOU, in the case it is fully repaid

Anyway, can I skip this part here? I task from module 4 (cash flow) we have option to consume some input and generate output, so for sure I will do lot of mistakes there, maybe no need to do it twice? ;)  
Unless there is something tricky here and it's worth to spend time to implement it here?

Reply:
Fair point. Make sure you allow an atomic cash repay operation along with IOU destroy, and when this is done, the "send money along with IOU create" will be a cakewalk.